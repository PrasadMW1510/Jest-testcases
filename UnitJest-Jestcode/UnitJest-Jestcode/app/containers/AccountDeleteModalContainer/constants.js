/*
 *
 * AccountDeleteModalContainer constants
 *
 */

export const POST_ACCOUNT_DELETE_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST';
export const POST_ACCOUNT_DELETE_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_FAILURE';
export const POST_ACCOUNT_DELETE_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_SUCCESS';
export const POST_ACCOUNT_DELETE_MIA_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_MIA_DELETE_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_FAILURE';
export const POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS';
