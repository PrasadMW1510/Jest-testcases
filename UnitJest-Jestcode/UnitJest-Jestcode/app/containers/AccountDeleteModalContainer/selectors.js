import { createSelector } from 'reselect';

/**
 * Direct selector to the accountDeleteModalContainer state domain
 */
const selectAccountDeleteModalContainerDomain = state => state.get('accountDeleteModalContainer');

/**
 * Default selector used by AccountDeleteModalContainer
 */

const makeSelectAccountDeleteModalContainer = () =>
  createSelector(selectAccountDeleteModalContainerDomain, substate => substate.toJS());

export default makeSelectAccountDeleteModalContainer;
export { selectAccountDeleteModalContainerDomain };
