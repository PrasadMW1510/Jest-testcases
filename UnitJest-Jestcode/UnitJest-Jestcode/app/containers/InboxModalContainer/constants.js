/*
 *
 * InboxModalContainer constants
 *
 */

export const DEFAULT_ACTION = 'app/InboxModalContainer/DEFAULT_ACTION';
export const GET_STUDENT_PROGRAM_DETAIL_REQUEST =
  'containers/InboxModalContainer/get_student_program_detail_request';
export const GET_STUDENT_PROGRAM_DETAIL_REQUEST_SUCCESS =
  'containers/InboxModalContainer/get_student_program_detail_request_success';
export const GET_STUDENT_SKILL_ASSESTMENT_QUESTION =
  'containers/InboxModalContainer/get_student_assestment_question';
export const GET_STUDENT_SKILL_ASSESTMENT_QUESTION_SUCCESS =
  'containers/InboxModalContainer/get_student_assestment_question_success';
export const SAVE_STUDENT_EVALUATION_DATA =
  'containers/InboxModalContainer/save_student_evaluation_data';
export const SAVE_STUDENT_EVALUATION_DATA_SUCCESS =
  'containers/InboxModalContainer/save_student_evaluation_data_success';
export const GET_STUDENT_SKILL_ASSESTMENT_QUESTION_FAILURE =
  'containers/InboxModalContainer/get_student_assestment_question_failure';

export const COMMUNITY_ID = 'R180NG,S44NG,S44JR,M180,M180Y2';
export const UNINITIALIZED_ITEM_COUNT = 'containers/InboxModalContainer/uninitialized_item_count';
