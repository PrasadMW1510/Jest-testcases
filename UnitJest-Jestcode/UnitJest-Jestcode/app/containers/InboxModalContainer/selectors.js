import { createSelector } from 'reselect';

/**
 * Direct selector to the inboxModalContainer state domain
 */
const selectInboxModalContainerDomain = state => state.get('inboxModalContainer');

/**
 * Other specific selectors
 */

/**
 * Default selector used by InboxModalContainer
 */

const makeSelectInboxModalContainer = () =>
  createSelector(selectInboxModalContainerDomain, substate => substate.toJS());

export default makeSelectInboxModalContainer;
export { selectInboxModalContainerDomain };
