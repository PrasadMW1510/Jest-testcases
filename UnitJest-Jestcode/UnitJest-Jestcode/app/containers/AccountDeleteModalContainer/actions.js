/*
 *
 * AccountDeleteModalContainer actions
 *
 */
import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteRequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteMIARequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_MIA_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete is a success
 *
 *  @returns {{type}}
 */
export function postAccountDeleteRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account delete is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountDeleteRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_FAILURE,
    error,
  });
}

/**
 * Handles API logic for account unenroll action
 *
 * @param accountUnenrollPayload
 * @returns {{type}}
 */
export function postAccountUnenrollRequest(accountUnenrollPayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST,
    payload: accountUnenrollPayload,
  };
}

/**
 * Handles API logic for account unenroll is a success
 *
 *  @returns {{type}}
 */
export function postAccountUnenrollRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account unenroll is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountUnenrollRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_FAILURE,
    error,
  });
}
