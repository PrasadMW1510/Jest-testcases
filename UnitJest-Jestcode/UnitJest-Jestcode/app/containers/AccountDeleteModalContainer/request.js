import API from 'utils/request';
import { COHORT_TYPE } from 'containers/App/constants';

export const postAccountsDelete = (sessionId, deletePayload) => {
  const result = API.post('/SlmsStudent', deletePayload, {
    params: {
      command: 'disable_if_unattached',
      sid: sessionId,
    },
  }).then(response => response.output_data[0]);
  return result;
};

export const postAccountsMIADelete = (sessionId, deletePayload, cohortType) => {
  let request = '';
  let command = 'disable_if_unattached';
  switch (cohortType) {
    case COHORT_TYPE.Student:
      request = '/SlmsStudent';
      break;
    case COHORT_TYPE.Teacher:
      request = '/SlmsTeacher';
      break;
    case COHORT_TYPE.Class:
      request = '/SlmsClass';
      command = 'delete';
      break;
    case COHORT_TYPE.School:
      request = '/SlmsSchool';
      command = 'delete';
      break;
    default:
      request = '/SlmsStudent';
      break;
  }

  const result = API.post(request, deletePayload, {
    params: {
      command,
      sid: sessionId,
    },
  }).then(response => (response.output_data[0] ? response.output_data[0] : response));

  return result;
};

export const postAccountsUnenroll = (sessionId, unenrollPayload) => {
  const result = API.post('/SlmsStudent', unenrollPayload, {
    params: {
      command: 'unenroll_if_unattached',
      sid: sessionId,
    },
  }).then(response => response.output_data[0]);
  return result;
};
