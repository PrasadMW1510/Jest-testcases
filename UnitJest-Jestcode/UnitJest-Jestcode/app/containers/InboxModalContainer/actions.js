/*
 *
 * InboxModalContainer actions
 *
 */

import * as Constants from './constants';

export function getStudentProgramDetailsDataRequest(programdata) {
  return {
    type: Constants.GET_STUDENT_PROGRAM_DETAIL_REQUEST,
    programdata,
  };
}

export function getStudentProgramDetailsDataSuccess(resultsData) {
  return {
    type: Constants.GET_STUDENT_PROGRAM_DETAIL_REQUEST_SUCCESS,
    resultsData,
  };
}

export function getQuestion(path) {
  return {
    type: Constants.GET_STUDENT_SKILL_ASSESTMENT_QUESTION,
    path,
  };
}

export function getStudentQuestionDataSuccess(data) {
  return {
    type: Constants.GET_STUDENT_SKILL_ASSESTMENT_QUESTION_SUCCESS,
    data,
  };
}

export function getStudentQuestionDataFailure(data) {
  return {
    type: Constants.GET_STUDENT_SKILL_ASSESTMENT_QUESTION_FAILURE,
    data,
  };
}

export function storeEvaluationUpdate(postdata, evData) {
  return {
    type: Constants.SAVE_STUDENT_EVALUATION_DATA,
    postdata,
    evData,
  };
}

export function saveStudentEvalulationDataSuccess(responseData) {
  return {
    type: Constants.SAVE_STUDENT_EVALUATION_DATA_SUCCESS,
    responseData,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}
