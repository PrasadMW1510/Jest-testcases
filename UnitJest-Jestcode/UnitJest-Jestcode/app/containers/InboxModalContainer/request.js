import API, { getBaseUrlWithoutSlms, getSkillQuestionURL } from 'utils/request';
import Axios from 'axios';
import * as Constants from './constants';

/**
 * Get Profile Details for District Admin
 *
 * @param sessionId
 * @param districtId
 */
export const getStudentProgramDetailsDataRequestData = (searchopts, sessionId, userId, userType) =>
  API.post(`${getBaseUrlWithoutSlms()}/slms/SlmsSdp`, searchopts, {
    params: {
      sid: sessionId,
      command: 'get_student_submissions',
      communityIds: Constants.COMMUNITY_ID,
      cohort_id: userId,
      cohort_type: userType,
    },
  }).then(response => response);

export const getStudentQuestionRequestData = filepath =>
  Axios.get(`${getSkillQuestionURL()}/slms-static/dev-2017-bts/${filepath}`, {
    params: {},
  }).then(response => response);

export const saveStudentEvaluationDataRequest = (sessionId, appID, postparams) =>
  API.post(`${getBaseUrlWithoutSlms()}/slms/SlmsSdp`, postparams, {
    params: {
      command: 'evaluate_student_submissions',
      app_id: appID,
      sid: sessionId,
    },
  }).then(response => response);
