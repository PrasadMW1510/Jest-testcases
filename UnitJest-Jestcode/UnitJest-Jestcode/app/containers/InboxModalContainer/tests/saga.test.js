import { call, put, takeLatest } from 'redux-saga/effects';
import { showLoading, hideLoading } from 'containers/LoadingController/actions';
import { fromJS } from 'immutable';
import defaultSaga, * as Saga from '../saga';

import * as Selectors from '../../App/selectors';
import * as Actions from '../actions';
import * as Request from '../request';
import * as Constants from '../constants';
/* eslint-disable redux-saga/yield-effects */
describe('InboxModalContainer Saga', () => {
  let generator = null;
  let loginDataSelector = null;
  let profileSessionIdSelector = null;
  let profileUserIdSelector = null;
  let profileUserTypeSelector = null;
  beforeEach(() => {
    profileSessionIdSelector = jest.fn();
    profileUserIdSelector = jest.fn();
    loginDataSelector = jest.fn().mockReturnValue(fromJS({ session_id: ['abdecf'] }));
    profileUserTypeSelector = jest.fn();
    jest.spyOn(Selectors, 'makeSelectLoginData').mockReturnValue(loginDataSelector);
    jest.spyOn(Selectors, 'makeSelectProfileSessionId').mockReturnValue(profileSessionIdSelector);
    jest.spyOn(Selectors, 'makeSelectProfileUserId').mockReturnValue(profileUserIdSelector);
    jest.spyOn(Selectors, 'makeSelectProfileUserType').mockReturnValue(profileUserTypeSelector);
  });

  describe('getStudentQuestionDataRequest', () => {
    const filepath = {
      path: ['inbox'],
    };
    beforeEach(() => {
      generator = Saga.getStudentQuestionDataRequest(filepath);
    });

    it('getSearchResultsRequestsuccess', () => {
      expect(generator.next().value).toEqual(
        call(Request.getStudentQuestionRequestData, filepath.path[0])
      );
      expect(generator.next(filepath).value).toEqual(
        put(Actions.getStudentQuestionDataSuccess(filepath))
      );
    });

    it('getSearchResultsRequestFailure', () => {
      const error = 'mock error';
      expect(generator.next().value).toEqual(
        call(Request.getStudentQuestionRequestData, filepath.path[0])
      );
      expect(generator.throw(error).value).toEqual(
        put(Actions.getStudentQuestionDataFailure(error))
      );
    });
  });

  describe('saveStudentEvaluationData', () => {
    const inputparams = {
      evData: {
        communityID: 'kjdn_jsa',
      },
      postdata: 'kjkn_nbxb',
    };
    beforeEach(() => {
      generator = Saga.saveStudentEvaluationData(inputparams);
    });

    it('getSearchResultsRequestsuccess', () => {
      expect(generator.next().value).toEqual(put(showLoading()));
      expect(generator.next('SessionIDVal').value).toMatchSnapshot();
      expect(generator.next().value).toEqual(
        call(
          Request.saveStudentEvaluationDataRequest,
          undefined,
          inputparams.evData.communityID,
          inputparams.postdata
        )
      );
      expect(generator.next().value).toEqual(put(Actions.saveStudentEvalulationDataSuccess()));
      expect(generator.next().value).toEqual(put(hideLoading()));
    });
    it('getSearchResultsRequestfailure', () => {
      const error = 'mock error';
      expect(generator.next().value).toEqual(put(showLoading()));
      expect(generator.next('SessionIDVal').value).toMatchSnapshot();
      expect(generator.throw(error).value).toEqual(put(hideLoading()));
    });
  });

  describe('getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined', () => {
    let mockSearchOpts;
    let store = null;
    beforeEach(() => {
      store = fromJS({
        login: {
          session_id: ['adsfczas111'],
          user_id: ['user123'],
          user_type: ['teacher'],
        },
        profile: {
          school_id: ['my_school'],
        },
      });
      mockSearchOpts = {
        programdata: {},
      };
      generator = Saga.getStudentProgramDetailsDataRequest(mockSearchOpts);
    });

    it('should successfully run', () => {
      const serviceResponse = {
        output_data: [
          {
            workItems: [
              {
                mSkillsWorkItem: [
                  {
                    mSkillsQuestion1: [
                      {
                        question: 'question1',
                      },
                    ],
                    mSkillsQuestion2: [
                      {
                        question: 'question2',
                      },
                    ],
                    mSkillsQuestion3: [
                      {
                        question: 'question3',
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      };
      const mockQuestions = ['question1Answer', 'question2Answer', 'question3Answer'];
      expect(generator.next().value).toEqual(put(showLoading()));
      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login', 'session_id', 0])).value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login', 'user_id', 0])).value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login'])).value).toEqual(
        call(
          Request.getStudentProgramDetailsDataRequestData,
          mockSearchOpts.programdata,
          store.getIn(['login', 'session_id', 0]),
          store.getIn(['login', 'user_id', 0]),
          store.getIn(['login', 'user_type', 0])
        )
      );
      expect(generator.next(serviceResponse).value).toEqual(
        put(Actions.getStudentProgramDetailsDataSuccess(serviceResponse))
      );

      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next('question1Answer').value).toMatchSnapshot();
      expect(generator.next('question2Answer').value).toMatchSnapshot();
      expect(generator.next('question3Answer').value).toEqual(
        put(Actions.getStudentQuestionDataSuccess(mockQuestions))
      );
      expect(generator.next().value).toEqual(put(hideLoading()));
    });

    it('should successfully run with Question3 undefined', () => {
      const serviceResponse = {
        output_data: [
          {
            workItems: [
              {
                mSkillsWorkItem: [
                  {
                    mSkillsQuestion1: [
                      {
                        question: 'question1',
                      },
                    ],
                    mSkillsQuestion2: [
                      {
                        question: 'question2',
                      },
                    ],
                    mSkillsQuestion3: undefined,
                  },
                ],
              },
            ],
          },
        ],
      };
      const mockQuestions = ['question1Answer', 'question2Answer'];
      expect(generator.next().value).toEqual(put(showLoading()));
      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login', 'session_id', 0])).value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login', 'user_id', 0])).value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login'])).value).toEqual(
        call(
          Request.getStudentProgramDetailsDataRequestData,
          mockSearchOpts.programdata,
          store.getIn(['login', 'session_id', 0]),
          store.getIn(['login', 'user_id', 0]),
          store.getIn(['login', 'user_type', 0])
        )
      );
      expect(generator.next(serviceResponse).value).toEqual(
        put(Actions.getStudentProgramDetailsDataSuccess(serviceResponse))
      );
      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next('question1Answer').value).toMatchSnapshot();
      expect(generator.next('question2Answer').value).toEqual(
        put(Actions.getStudentQuestionDataSuccess(mockQuestions))
      );
      expect(generator.next().value).toEqual(put(hideLoading()));
    });

    it('should unsuccessfully run with Question is null', () => {
      const serviceResponse = {
        output_data: [
          {
            workItems: [
              {
                mSkillsWorkItem: [
                  {
                    mSkillsQuestion1: [
                      {
                        question: 'question1',
                      },
                    ],
                    mSkillsQuestion2: null,
                    mSkillsQuestion3: [
                      {
                        question: 'question3',
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      };

      expect(generator.next().value).toEqual(put(showLoading()));
      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login', 'session_id', 0])).value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login', 'user_id', 0])).value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login'])).value).toEqual(
        call(
          Request.getStudentProgramDetailsDataRequestData,
          mockSearchOpts.programdata,
          store.getIn(['login', 'session_id', 0]),
          store.getIn(['login', 'user_id', 0]),
          store.getIn(['login', 'user_type', 0])
        )
      );
      expect(generator.next(serviceResponse).value).toEqual(
        put(Actions.getStudentProgramDetailsDataSuccess(serviceResponse))
      );
      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next('question1Answer').value).toEqual(put(hideLoading()));
    });

    it('should unsuccessfully run with service response mSkillsWorkItem null', () => {
      const serviceResponse = {
        output_data: [
          {
            workItems: [
              {
                mSkillsWorkItem: '',
              },
            ],
          },
        ],
      };
      expect(generator.next().value).toEqual(put(showLoading()));
      expect(generator.next().value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login', 'session_id', 0])).value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login', 'user_id', 0])).value).toMatchSnapshot();
      expect(generator.next(store.getIn(['login'])).value).toEqual(
        call(
          Request.getStudentProgramDetailsDataRequestData,
          mockSearchOpts.programdata,
          store.getIn(['login', 'session_id', 0]),
          store.getIn(['login', 'user_id', 0]),
          store.getIn(['login', 'user_type', 0])
        )
      );
      expect(generator.next(serviceResponse).value).toEqual(
        put(Actions.getStudentProgramDetailsDataSuccess(serviceResponse))
      );
      expect(generator.next().value).toEqual(put(hideLoading()));
      expect(generator.next().value).toMatchSnapshot();
    });
  });

  describe('getStudentProgramDetailsDataRequest when Error', () => {
    let mockSearchOpts;
    const err = 'mock error';
    beforeEach(() => {
      mockSearchOpts = {
        programdata: {},
      };
      generator = Saga.getStudentProgramDetailsDataRequest(mockSearchOpts);
    });

    it('should throw error', () => {
      expect(generator.next().value).toEqual(put(showLoading()));
      expect(generator.next('sessionID').value).toMatchSnapshot();
      expect(generator.next('userID').value).toMatchSnapshot();
      expect(generator.next('userType').value).toMatchSnapshot();
      expect(generator.throw(err).value).toEqual(put(hideLoading()));
    });
  });

  describe('defaultSaga', () => {
    beforeAll(() => {
      generator = defaultSaga();
    });

    it('Expects default Sagas are called', () => {
      expect(generator.next().value).toEqual(
        takeLatest(
          Constants.GET_STUDENT_PROGRAM_DETAIL_REQUEST,
          Saga.getStudentProgramDetailsDataRequest
        )
      );
    });

    it('Expects default Sagas are called', () => {
      expect(generator.next().value).toEqual(
        takeLatest(
          Constants.GET_STUDENT_SKILL_ASSESTMENT_QUESTION,
          Saga.getStudentQuestionDataRequest
        )
      );
    });

    it('Expects default Sagas are called', () => {
      expect(generator.next().value).toEqual(
        takeLatest(Constants.SAVE_STUDENT_EVALUATION_DATA, Saga.saveStudentEvaluationData)
      );
    });
  });
});
