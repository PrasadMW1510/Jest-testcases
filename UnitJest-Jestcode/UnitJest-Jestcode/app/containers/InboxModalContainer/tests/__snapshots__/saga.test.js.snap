// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when Error should throw error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when Error should throw error 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when Error should throw error 3`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run 3`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run 4`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "question1",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run 5`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "question2",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run 6`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "question3",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run with Question3 undefined 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run with Question3 undefined 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run with Question3 undefined 3`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run with Question3 undefined 4`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "question1",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should successfully run with Question3 undefined 5`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "question2",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should unsuccessfully run with Question is null 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should unsuccessfully run with Question is null 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should unsuccessfully run with Question is null 3`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should unsuccessfully run with Question is null 4`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "question1",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should unsuccessfully run with service response mSkillsWorkItem null 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should unsuccessfully run with service response mSkillsWorkItem null 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should unsuccessfully run with service response mSkillsWorkItem null 3`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga getStudentProgramDetailsDataRequest when mSkillsWorkItem is undefined should unsuccessfully run with service response mSkillsWorkItem null 4`] = `undefined`;

exports[`InboxModalContainer Saga saveStudentEvaluationData getSearchResultsRequestfailure 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`InboxModalContainer Saga saveStudentEvaluationData getSearchResultsRequestsuccess 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;
